cdap-kubernetes/src/main/java/io/cdap/cdap/k8s/runtime/KubeTwillPreparer.java

```diff
@@ -453,9 +456,10 @@ V1Deployment buildDeployment(RunId runid,
 
     V1ResourceRequirements resourceRequirements = new V1ResourceRequirements();
     Map<String, Quantity> quantityMap = new HashMap<>();
-    quantityMap.put("cpu", new Quantity(String.valueOf(vCores)));
+    quantityMap.put("cpu", new Quantity(String.valueOf((float) (vCores * kubeScalingFactors.get("cpu")))));
```

Comment:
Does `Quantity` takes floating point value? Should we round it up/down?

Reply:
Quantity converts the string number to BigDecimal so can be float. Ref: https://github.com/kubernetes-client/java/blob/master/kubernetes/src/main/java/io/kubernetes/client/custom/Quantity.java#L17

The default value of vCores is 1. Rounding up or down would remove the scaling effect we want here. Ref: https://github.com/apache/twill/blob/master/twill-api/src/main/java/org/apache/twill/api/ResourceSpecification.java#L28